### DISCUSSION TITLE
Escape as global reset?

### DISCUSSION THREAD

**Discussion by askreet** [helix-editor/helix#5703] [Ideas] [answered]
Apologies if this has been discussed elsewhere, I didn't find a previous thread.

I've been using Helix on-and-off for a while and I love it. The mental model of the modal shortcuts out-of-the-box really works for me in a way that feels fresh, with _one_ exception. I've had such a hard time reasoning about resetting my selection vs. clearing multiple cursor.

A video I saw on YouTube about Helix recently the host suggested this configuration:

```
[keys.normal]
esc = ["keep_primary_selection", "collapse_selection"]
```

For me, this really aligns with my mental model of the escape key in a modal editor: "get me back to normal!"

I'm curious if others have tried this and, if it's common at all, if it should be the default behavior? Perhaps I'm missing out on being able to keep selections and/or cursors while transitioning in/out of normal mode.

**Comment by the-mikedavis** [comment, marked as answer]
I think this binding makes sense if you mostly use multiple selections for small well-contained edits. If you're using multiple selections for large edits (a 20+ command string), this makes it really easy to lose your selections especially if you're entering and exiting insert mode often.

The current bindings for these are very easy to hit - `,` and `;` respectively - and I would prefer that these be explicit actions. Using Escape for this also makes it seem like multiple selections are a mode like insert/normal/select mode which I think is potentially confusing.

Also see https://github.com/helix-editor/helix/pull/5289

**Reply by askreet** [reply]
Thanks for the context, glad to see this has come up before. I agree with the confusion, I may try to spend more time learning the default bindings but mentally have such a hard time with using, for example, `f` to get to a period I want to delete, then needing to clear the created selection and press `d`. I'm sure it comes with time, though.
